### PR TITLE
fix: Simplify MyC local image logic

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [CX-3337] 

### Description

This PR simplifies MyC local image logic by storing the images locally by image ID instead of artwork ID.

Sorry for the large PR...

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Simplify MyC local image logic - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3337]: https://artsyproduct.atlassian.net/browse/CX-3337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Screens/ArtworkList/MyCollectionArtworkGridItem.tsx [hunk 1]
```diff
@@ -51,6 +49,13 @@ const MyCollectionArtworkGridItem: React.FC<MyCollectionArtworkGridItemProps> =
 
   const showHighDemandIcon = isP1Artist && isHighDemand
 
+  const [timer, setTimer] = useState(0)
+  useEffect(() => {
+    setInterval(() => {
+      setTimer(timer + 1)
+    })
+  }, [])
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Screens/ArtworkList/MyCollectionArtworkGridItem.tsx:57]: **Suggestion**: Maybe we should clear the interval on the end of the useEffect?
```suggestion
  useEffect(() => {
    const interval = setInterval(() => {
      setTimer(timer + 1)
    })
    return () => {
    	clearInterval(interval)
    }
  }, [])
```

**Review by MrSltun — APPROVED** [review]
LGTM! ✨ Thank you for this great work Ole!! 🌟 🚀 
I just left a few small comments :)

File: src/app/Scenes/Artwork/Components/ImageCarousel/geometry.ts [hunk 2]
```diff
@@ -78,12 +78,12 @@ export function getMeasurements({
...
     result.push({
-      width,
-      height,
-      marginLeft: marginLeft + paddingLeft,
-      marginRight: marginHorizontal,
-      marginTop: marginVertical,
-      marginBottom: marginVertical,
+      width: width || 0,
+      height: height || 0,
+      marginLeft: marginLeft + paddingLeft || 0,
```
[the remark this answers is not in the document]
Comment (olerichter00) [reply src/app/Scenes/Artwork/Components/ImageCarousel/geometry.ts:83]: `+` has priority before `||` and both versions are equivalent and my editor auto-fixes the line to `marginLeft: marginLeft + paddingLeft || 0,` 🤷

[the remark this answers is not in the document]
Comment (MrSltun) [reply src/app/Scenes/Artwork/Components/ImageCarousel/geometry.ts:83]: Oh in that case, please ignore this comment 😄

File: src/app/Scenes/MyCollection/Screens/ArtworkList/MyCollectionArtworkGridItem.tsx (hunk 1 above)
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Screens/ArtworkList/MyCollectionArtworkGridItem.tsx:57]: Good catch! Actually I used the code only for debugging purposes 🤦 Will remove it :)

**Pull request merged**